security/pkg/pki/ca/ca.go

```diff
@@ -383,3 +389,24 @@ func (ca *IstioCA) GenKeyCert(hostnames []string, certTTL time.Duration) ([]byte
 
 	return certPEM, privPEM, nil
 }
+
+// if CA cert becomes invalid before workload cert it's going to cause workload cert to be invalid too,
+// however citatel won't rotate if that happens, this function will prevent that using cert chain TTL as
+// the workload TTL
+func (ca *IstioCA) getDefaultCertTTL(defaultCertTTL time.Duration) (time.Duration, error) {
+	certChainPem := ca.keyCertBundle.GetCertChainPem()
+	if len(certChainPem) == 0 {
+		return defaultCertTTL, nil
+	}
+
+	certChainTTL, err := util.TimeBeforeCertExpires(certChainPem, time.Now())
+	if err != nil {
+		return defaultCertTTL, fmt.Errorf("failed to get cert chain TTL %s", err.Error())
+	}
+
+	if defaultCertTTL.Seconds() > certChainTTL.Seconds() {
+		return certChainTTL, nil
```

Comment:
Since certChainTTL returned by util.TimeBeforeCertExpires() can be negative, the TTL value returned by getDefaultCertTTL() can be negative too, which is invalid.